packages/qdrant-loader-core/tests/llm/providers/test_bedrock.py

```
@@ -0,0 +1,368 @@
+
+    with pytest.raises(import_module("qdrant_loader_core.llm.errors").ServerError):
+        await provider.embeddings().embed(["hello"])
+
+
+def test_bedrock_import_fallback_with_missing_boto3(monkeypatch):
+    module_name = "qdrant_loader_core.llm.providers.bedrock"
+    original_module = sys.modules.pop(module_name, None)
+
+    original_import = builtins.__import__
+
+    def fake_import(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `fake_import`

(ANN202)